Remove UrlRedirectResolver

RoiEXLab opened the pull request.

DanVanAtta commented:
Looks good, @RoiEXLab  can you confirm the testing that was done to verify this?

RoiEXLab commented:
Haven't had the time to properly test this, but it should work unless httpcomponents is broken... 
FileDownloader#download is using HTTPComponents to download files. The client object is being retrieved by `newHttpClient()` which I modified to return a handler which follows redirects.
The LaxRedirectHandler follows every kind of redirect, but has an issue where Request bodies are not being re-sent on 301 permanent redirects

RoiEXLab commented:
I'm going to test this soon though 👍

ssoloff commented on src/main/java/games/strategy/engine/framework/map/download/DownloadUtils.java:
```diff
@@ -102,6 +103,7 @@ private DownloadUtils() {}
   private static CloseableHttpClient newHttpClient() {
     return HttpClients.custom()
         .setDefaultRequestConfig(RequestConfig.custom().setCookieSpec(CookieSpecs.IGNORE_COOKIES).build())
+        .setRedirectStrategy(new LaxRedirectStrategy())
```
I'm curious why `DefaultRedirectStrategy` (the default redirect strategy used by `HttpClientBuilder`) is not sufficient here?  `LaxRedirectStrategy` seems to only be necessary when the request is a POST or PUT.  This class only ever makes `GET` and `HEAD` requests, which appear to be handled correctly by `DefaultRedirectStrategy`.

BTW, the code in this class was tested to follow redirects of `GET` and `HEAD` requests as part of the testing for #1718.

RoiEXLab replied:
Agreed

ssoloff replied:
Sorry, I should have been clearer in my previous comment.  The line configuring `DefaultRedirectStrategy` is not necessary because `HttpClientBuilder` will use a `DefaultRedirectStrategy` if a custom strategy is not specified.  NBD; functionality is the same. :smile:

RoiEXLab commented:
@DanVanAtta @ssoloff Updated!
Everything works as expected, I changed the `lobby_properties_file_url`to a goo.gl shortened link and removed the lobby backup url.
I could still connect to the lobby
